podman_hpc/hook_tool.py

```diff
@@ -62,10 +76,67 @@ def ldconfig():
         log(f"ldconfig failed: {ret}")
 
 
-def resolve_src(src, modulesd=os.path.abspath("")):
-    if not os.path.isabs(src):
-        src = os.path.join(modulesd, src)
-    return os.path.abspath(os.path.expandvars(src))
+def resolve_src_and_dest(rule, root_path, modulesd=os.path.abspath("")):
+    # extract source and destination patterns from the rule
+    rs, rd = (rule + ":").split(":")[:2]
+
+    # expand vars
+    rs = os.path.expanduser(os.path.expandvars(rs))
+    rd = os.path.expanduser(os.path.expandvars(rd))
+
+    # ensure source pattern is absolute path
+    if not os.path.isabs(rs):
+        rs = os.path.join(modulesd, rs)
+    rs = os.path.abspath(os.path.expanduser(os.path.expandvars(rs)))
+
+    # error checks
+    if not os.path.isabs(rd):
+        log(
+            f"Error: Destination in pattern must be an absolute path.\n\tdestination: {rd}"
+        )
+        return {}
+    if "*" in rd:
+        if rs.count("*") != 1:
+            log(
+                "Error: Using glob '*' in destination requires exactly one glob '*' in source."
+                f"\n\tsource: {rs}\n\tdestination: {rd}"
+            )
+            return {}
+        else:
+            # this is used to capture the glob expansion later
+            globstrip = "^{}|{}$".format(*rs.split("*"))
+
+    # determine how to construct absolute path to destination object
+    # if no destination rule is given, use the path from source rule
+    if rd == "":
+        dest = lambda src: os.path.join(root_path, src[1:])
```

Comment:
where are lambdas being used?

Reply:
The dictionary comprehension in the return line.